### ISSUE TITLE
WYSIWYG Editor 

### ISSUE THREAD

**Issue by MDXDave** [zammad/zammad#330]
The example triggers includes bold text, but it is not possible to style your own text (except this workaround)
![zammad2](https://cloud.githubusercontent.com/assets/6099976/19830729/10704f24-9dfe-11e6-98a0-d173b0d614a6.gif)

How about include a _simple_ WYSIWYG editor including bold, italic, underlinde, font size and images.

**Comment by martini** [comment]
Note: It is also requested in T#1082152


Hi @MDXDave 

did you see the ctrl+alt+h shortcuts? Would this be fine for simple WYSIWYG editor?

<img width="739" alt="screen shot 2016-10-29 at 18 27 14" src="https://cloud.githubusercontent.com/assets/110226/19831248/68389c5a-9e05-11e6-8961-8de61ea7bfe1.png">

**Comment by MDXDave** [comment]
Oh i have missed this. That is quite enough.

But maybe it would be easier to use with a toolbar like here in github?

**Comment by martini** [comment]
Yeah, currently it's intend to keep the UI clean as possible - but if you miss features like this... I'll discuss it with our designers.

PS: With the short cuts you can also close tabs, it's really nice to increase efficiency.

**Comment by Julian0o** [comment]
The shortcut system is nice! No need for me to have a WYSIWYG editor.

**Comment by MartinBorg** [comment]
New users either need some kind of buttons to change style (because they do not even know about short cuts or can't remember them).

Or there should be Zammad coffee mugs with editor shortcuts. Problem: I'm an espresso fanatic, so there is space only for one or two shortcuts.

Btw. I like the way Trac does editing with easy changing between their wiki markup editing and WYSIWYG. Includes tables etc.

**Comment by AND51** [comment]
Hello,

I would like to share my opinion with you. I know the shortcut sheet cheat (ctrl+alt+h). Rarely I use a shortcut to make text **bold**, so yes, I know this feature. But you mustn't forget that one cannot keep everything in mind. I also work with Zendesk every day and even if there is a RichText editor, I only use 2 shortcuts regularly (insert code tags and creating an bullet list). Everything else like bold, italic, insert a table and so on is being done so rare that I can simply not remember the corresponding shortcut. Please, compare this screenshot I took from Zendesk:

![screenshot-2017-04-16-000310](https://cloud.githubusercontent.com/assets/26226303/25067081/faa0b85e-2238-11e7-9077-ef254e54ea15.png)

In my opinion, a small extra bar doesn't hurt really. The UI still looks clean. Zendesk did it like this: the separate bar appears only when you click "T". Zammad may implement this eventually different, of course. Maybe there will even be an option in the settings where you can turn this feature on and off (to also satisfy people who use shortcuts only). In many cases, you select the text even with the mouse. Why switching back to keyboard then? If you already have the mouse in the hand, you can quickly click a button to format the text. Last, I admit that this issue depends on one's personal preferences heavily.

**Comment by ntinti** [comment]
I would like the possibility to paste images directly from the clipboard or to format text with fixed width like courier e.g. for log file inputs. tables also will be fine.

**Comment by ntinti** [comment]
Oh, I see pasting images with firefox is possible, but not with safari... But there should be more possibilities to format text. If you like a clean UI a markdown syntax would be a solution.

**Comment by passboltUser** [comment]
I had to search it and found this issue. It helped me, but if there were buttons I wouldn't need to search.

**Comment by flesser** [comment]
+1 for simple Markdown support!

No need for fancy buttons or any UI changes, just apply a standard Markdown parser on internal notes allowing for simple helpers like `**bold**`, `_italic_` and especially preformatted `'code'`. Maybe some `* lists` would be nice, too.

**Comment by cheoAlejo** [comment]
+1 for markdown.

**Comment by MrMontesa** [comment]
+1 for markdown.
Looking for a solution to have 'code' or 'command line output' styled nicely in internal ticket updates.

**Comment by incognitozen** [comment]
+1 on a rich editor. The Github editor is ideal.

**Comment by lcx** [comment]
+1 on markdown. Already using lot's of markdown, I could live with the existing solution if there was something for code

**Comment by contaoacademy** [comment]
+1 markdown

**Comment by MrGeneration** [comment]
Thanks for your interest in this enhancement. Please refrain from replying on github issues with just "+1 [...]", as this generates quite some noise to people watching this repository. ~~Instead, please use a thumb up on the intial post. That's much better. :)~~ 🚀

**Comment by dvuckovic** [comment]
With Zammad 7.0, a small unobtrusive bubble menu will come for the legacy editor (activated by selecting text first):

<img width="711" height="153" alt="Image" src="https://github.com/user-attachments/assets/8e3d3561-38a7-48b2-9e82-c0b38c9ebf75" />

For desktop view, we will have a proper content editor now with many more features and permanent "sticky" toolbar:

<img width="1035" height="176" alt="Image" src="https://github.com/user-attachments/assets/7f8c73ad-e021-498f-a31b-d81cf6158f29" />
